crates/swc_ecma_lints/src/config.rs

```diff
@@ -1,44 +1,61 @@
-use crate::rules::no_console::NoConsoleConfig;
-use serde::{Deserialize, Serialize};
-use std::fmt::Debug;
-
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
-#[serde(rename_all = "lowercase")]
-pub enum LintRuleReaction {
-    Off,
-    Warning,
-    Error,
-}
+#[cfg(feature = "non_critical_lints")]
+#[path = ""]
+mod non_critical_lints {
+    use crate::rules::non_critical_lints::no_console::NoConsoleConfig;
+    use serde::{Deserialize, Serialize};
+    use std::fmt::Debug;
 
-impl Default for LintRuleReaction {
-    fn default() -> Self {
-        Self::Off
+    #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+    #[serde(rename_all = "lowercase")]
+    pub enum LintRuleReaction {
```

Comment:
I think this type should exist regardless of features.
The only things we need to feature-gate is impl itself (module) and the function `all`.